client/state/jetpack-connect/reducer.js

```diff
@@ -264,6 +265,29 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 	return state;
 }
 
+export function jetpackAuthAttempts( state = {}, action ) {
+	switch ( action.type ) {
+		case JETPACK_CONNECT_RETRY_AUTH:
+			const slug = action.slug;
+			let currentTimestamp = state[ slug ] ? state[ slug ].timestamp || Date.now() : Date.now();
+			let attemptNumber = action.attemptNumber;
+			if ( attemptNumber > 0 ) {
+				const now = Date.now();
+				if ( now - currentTimestamp > AUTH_ATTEMPS_TTL ) {
+					currentTimestamp = now;
+					attemptNumber = 0;
+				}
+			}
+			return Object.assign( {}, state, { [ slug ]: { attempt: attemptNumber, timestamp: currentTimestamp } } );
+		case JETPACK_CONNECT_COMPLETE_FLOW:
+			return {};
+		case DESERIALIZE:
```

Comment:
Hmm, not agree. Having in mind that each connection attempt can take quite some time, I think checking if the data is stale when the actual event is triggered is a better approach than just checking it when calypso is loaded (but we could add that check in addition too, of course)